### PR TITLE
scaffolder: Split out Actions into smaller modules

### PR DESCRIPTION
**Pull request by benjdlambert**
Implementation of #21442

We're wanting to split apart the scaffolder and break apart the actions into seperate `-github` `-azure` `-bitbucket` etc actions so that we can have a better ownership model of these modules going forward.

### CODE REVIEW

File: plugins/scaffolder-backend-module-gitlab/package.json [hunk 1]
```diff
@@ -31,16 +31,18 @@
...
   },
   "dependencies": {
+    "@backstage/backend-common": "workspace:^",
     "@backstage/config": "workspace:^",
     "@backstage/errors": "workspace:^",
     "@backstage/integration": "workspace:^",
     "@backstage/plugin-scaffolder-node": "workspace:^",
-    "@gitbeaker/node": "^35.8.0",
+    "@gitbeaker/core": "^35.8.0",
+    "@gitbeaker/node": "^35.1.0",
```
Comment (freben) [inline plugins/scaffolder-backend-module-gitlab/package.json:40]: was this intentional? you downgraded the node version range

File: plugins/scaffolder-node/src/actions/gitHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ * limitations under the License.
+ */
+
+import { Git } from '@backstage/backend-common';
+import { Logger } from 'winston';
+
+/**
+ * @public
+ */
+export async function initRepoAndPush({
```
Comment (freben) [inline plugins/scaffolder-node/src/actions/gitHelpers.ts:23]: should not unpack the options in the function head

File: plugins/scaffolder-node/src/actions/gitHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    remote: 'origin',
+  });
+
+  return { commitHash };
+}
+
+/**
+ * @public
+ */
+export async function commitAndPushRepo({
```
Comment (freben) [inline plugins/scaffolder-node/src/actions/gitHelpers.ts:84]: should not unpack the options in the function head

**Review by freben — COMMENTED** [review]
Almost perfect! 🥳

**Review by Rugvip — APPROVED** [review]
Pristine 👌 :shipit:

**Pull request merged**